### PR TITLE
Made vision drive

### PR DESCRIPTION
**Pull request by NWSchallock**
PTAL

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -12,5 +12,48 @@
     ".project": true
   },
   "C_Cpp.default.configurationProvider": "vscode-wpilib",
-  "wpilib.online": true
```
Comment (duane701) [inline .vscode/settings.json]: Should `settings.json` be checked in? How did all this information end up in this file?

File: src/main/include/subsystems/Chassis.h [hunk 2]
```diff
@@ -49,6 +49,7 @@ class Chassis: public frc::Subsystem {
   double GetLeftVoltage();
   double GetRightVoltage();
   void SetArcadeDrive(double speed, double rotation);
+  double SetRotation();
```
Comment (duane701) [inline src/main/include/subsystems/Chassis.h]: Seems like `SetRotation` should be something like, `GetVisionRotation`.

File: src/main/cpp/subsystems/Chassis.cpp [hunk 3]
```diff
@@ -13,6 +15,7 @@ std::shared_ptr<Chassis> Chassis::getInstance() {
   return self;
 }
 
+std::shared_ptr<NetworkTable> table = nt::NetworkTableInstance::GetDefault().GetTable("limelight");
```
Comment (duane701) [inline src/main/cpp/subsystems/Chassis.cpp]: `table` cannot go here. Move table to a member variable (where it should be named `mLimelightTable` (or something similar)) of `Chassis` and initialize in the constructor.

**Comment by NWSchallock** [comment]
PTAL

**Comment by duane701** [comment]
I think you did not build this branch.

FTP

File: src/main/include/subsystems/Chassis.h [hunk 4]
```diff
@@ -49,7 +51,7 @@ class Chassis: public frc::Subsystem {
   double GetLeftVoltage();
   double GetRightVoltage();
   void SetArcadeDrive(double speed, double rotation);
-  double SetRotation();
+  double GetVisionRotation();
```
Comment (duane701) [inline src/main/include/subsystems/Chassis.h]: You renamed `GetVisionRotation` here. What about in `VisionDrive`?

File: src/main/cpp/subsystems/Chassis.cpp [hunk 5]
```diff
@@ -15,9 +15,6 @@ std::shared_ptr<Chassis> Chassis::getInstance() {
   return self;
 }
 
-std::shared_ptr<NetworkTable> table = nt::NetworkTableInstance::GetDefault().GetTable("limelight");
```
Comment (duane701) [inline src/main/cpp/subsystems/Chassis.cpp]: table was moved to `Chassis.hpp`. 

```
#include "networktables/NetworkTable.h"
#include "networktables/NetworkTableInstance.h"
```
Also needs to be moved. Also use `<>` for system includes.

File: src/main/include/subsystems/Chassis.h [hunk 6]
```diff
@@ -24,6 +24,8 @@ class Chassis: public frc::Subsystem {
   frc::SpeedControllerGroup m_right;
   frc::DifferentialDrive m_drive;
 
+  std::shared_ptr<NetworkTable> mLimeLightTable;
```
Comment (duane701) [inline src/main/include/subsystems/Chassis.h]: Use `nt:: NetworkTable` once you move the includes from the `.cpp`

File: src/main/cpp/subsystems/Chassis.cpp (hunk 5 above)
Comment (duane701) [reply src/main/cpp/subsystems/Chassis.cpp]: Are both of these needed in the `.h`? Perhaps only `#include "networktables/NetworkTable.h"` needs to move?

**Comment by duane701** [comment]
I confirmed

`#include <networktables/NetworkTableInstance.h>`

should stay in the `.cpp`

**Comment by NWSchallock** [comment]
PTAL i pretty sure it's good now

File: src/main/cpp/subsystems/Chassis.cpp [hunk 7]
```diff
@@ -1,6 +1,8 @@
 #include "subsystems/Chassis.h"
 #include "RobotMap.h"
 #include "commands/TankDrive.h"
+#include <networktables/NetworkTable.h>
```
Comment (duane701) [inline src/main/cpp/subsystems/Chassis.cpp]: `#include <networktables/NetworkTable.h>` is in the hpp file. Not needed here.

I'll fix this on the rebase.

**Pull request closed without merging**